Mage.Sets/src/mage/cards/t/TeferiWhoSlowsTheSunset.java

```diff
@@ -0,0 +1,135 @@
+package mage.cards.t;
+
+import mage.abilities.Ability;
+import mage.abilities.LoyaltyAbility;
+import mage.abilities.common.PlaneswalkerEntersWithLoyaltyCountersAbility;
+import mage.abilities.dynamicvalue.common.StaticValue;
+import mage.abilities.effects.OneShotEffect;
+import mage.abilities.effects.common.GetEmblemEffect;
+import mage.abilities.effects.common.LookLibraryAndPickControllerEffect;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.CardType;
+import mage.constants.Outcome;
+import mage.constants.SubType;
+import mage.constants.SuperType;
+import mage.filter.FilterCard;
+import mage.filter.common.FilterArtifactPermanent;
+import mage.filter.common.FilterCreaturePermanent;
+import mage.filter.common.FilterLandPermanent;
+import mage.game.Game;
+import mage.game.command.emblems.TeferiWhoSlowsTheSunsetEmblem;
+import mage.game.permanent.Permanent;
+import mage.players.Player;
+import mage.target.Target;
+import mage.target.TargetPermanent;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+/**
+ * @author LePwnerer
+ */
+public final class TeferiWhoSlowsTheSunset extends CardImpl {
+
+    public TeferiWhoSlowsTheSunset(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.PLANESWALKER}, "{2}{W}{U}");
+
+        this.addSuperType(SuperType.LEGENDARY);
+        this.subtype.add(SubType.TEFERI);
+        this.addAbility(new PlaneswalkerEntersWithLoyaltyCountersAbility(4));
+
+        // +1: Choose up to one target artifact, up to one target creature, and up to one target land. Untap the chosen permanents you control. Tap the chosen permanents you don't control. You gain 2 life.
+        this.addAbility(new LoyaltyAbility(new TeferiWhoSlowsTheSunsetEffect(), 1));
```

Comment:
this ability needs to have the targets added to is so they're chosen during activation and not during resolution